pkg/controller/certificate-shim/sync.go

```diff
@@ -257,11 +257,6 @@ func validateGatewayListenerBlock(path *field.Path, l gwapi.Listener) field.Erro
 	} else {
 		// check that each CertificateRef is valid
 		for i, secretRef := range l.TLS.CertificateRefs {
-			if secretRef == nil {
```

Comment:
Shouldn't we still be checking that e.g. the `name` field is set within the secretRef? It's no longer a pointer but it does have fields that we require to be set right? (or does some other kind of validation ensure these are set?)

Reply:
normally, these annotations on the `ObjectName` type should already enforce this:
```
// +kubebuilder:validation:MinLength=1
// +kubebuilder:validation:MaxLength=253
```

Reply:
https://github.com/kubernetes-sigs/gateway-api/blob/914a630b37a60a21f5f4fce3918633131dbef88d/apis/v1alpha2/shared_types.go#L403-L409

Reply:
I now remember why we had this validation in the first place. It is because cert-manager only supports Secret resources, and because we have decided to skip unsupported values in `certificateRefs`. In the following example, the first value is unsupported, and the second is supported, so it makes sense for cert-manager to "skip" the first one and to let the user know that the first item isn't supported.

```yaml
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: Gateway
spec:
  gatewayClassName: foo
  listeners:
    - name: http
      tls:
        certificateRefs:
          - name: example1-com-tls # ❌ Skipped because unsupported by cert-manager 
            kind: VaultSecret        
          - name: example2-com-tls # ✅ 
          - name: example2-com-tls # ✅ 
            kind: Secret             
          - name: example2-com-tls # ✅ 
            group: core
          - name: example2-com-tls # ✅ 
            group: core             
            group: Secret
```

Before Tim's change, we were checking whether the `secretRef` was a supported type or not. We assumed that an empty secretRef (i.e., when it is left nil) was unsupported by cert-manager.

So, given that we only want to filter out unsupported values in `certificateRefs`, the question is: what does cert-manager need to check in the `secretRef` in order to know whether it supports this `secretRef`?

I think it now only needs to check for the `kind` and `group`, like before. It doesn't need to check for the `name` to be able to know if this is a supported type.

On top of that, `name` is already validated by the openapi spec, as Tim mentioned.